server/pulp/server/async/app.py

```diff
@@ -49,8 +74,54 @@ def initialize_worker(sender, instance, **kwargs):
     """
     initialization.initialize()
 
+    # Delete any potential old state
     tasks._delete_worker(sender, normal_shutdown=True)
 
     # Create a new working directory for worker that is starting now
     common_utils.delete_worker_working_directory(sender)
     common_utils.create_worker_working_directory(sender)
+
```

Comment:
Everything you have added below is distinct from what happens above. It's a large body of code that operates independent of everything else in the function, so it's a great candidate for moving into a separate function.

One of the indicators I noticed that made me think about this was the stack of 7 mocks on one of the unit tests. That's often a sign that some re-organization would be beneficial, usually to separate parts of code that don't need to be together. Testing becomes much easier when the "units" get smaller.